lib/logstash/filters/geoip.rb

```diff
@@ -130,16 +154,36 @@ def filter(event)
         if value.is_a?(String)
           # Some strings from GeoIP don't have the correct encoding...
           value = case value.encoding
-            # I have found strings coming from GeoIP that are ASCII-8BIT are actually
-            # ISO-8859-1...
-            when Encoding::ASCII_8BIT; value.force_encoding(Encoding::ISO_8859_1).encode(Encoding::UTF_8)
-            when Encoding::ISO_8859_1, Encoding::US_ASCII;  value.encode(Encoding::UTF_8)
-            else; value
-          end
+                    # I have found strings coming from GeoIP that are ASCII-8BIT are actually
+                    # ISO-8859-1...
+                    when Encoding::ASCII_8BIT; value.force_encoding(Encoding::ISO_8859_1).encode(Encoding::UTF_8)
+                    when Encoding::ISO_8859_1, Encoding::US_ASCII;  value.encode(Encoding::UTF_8)
+                    else; value.dup
+                  end
```

Comment:
yes, the encode method returns a new string indeed so definitely no `dup` need there.